pyquil/device.py

```diff
@@ -113,158 +119,41 @@ def from_dict(d):
-        :return: The deserialized KrausModel.
-        :rtype: KrausModel
-        """
-        kraus_ops = [KrausModel.unpack_kraus_matrix(k) for k in d['kraus_ops']]
-        return KrausModel(d['gate'], d['params'], d['targets'], kraus_ops, d['fidelity'])
-
-    def __eq__(self, other):
-        return isinstance(other, KrausModel) and self.to_dict() == other.to_dict()
-
-    def __neq__(self, other):
-        return not self.__eq__(other)
-
-
-_NoiseModel = namedtuple("_NoiseModel", ["isa_name", "gates", "assignment_probs"])
-
-
-class NoiseModel(_NoiseModel):
-    """
-    Encapsulate the QPU noise model containing information about the noisy gates.
-
-    :ivar str isa_name: The name of the instruction set architecture for the QPU.
-    :ivar Sequence[KrausModel] gates: The tomographic estimates of all gates.
-    :ivar Dict[int,np.array] assignment_probs: The single qubit readout assignment
-        probability matrices keyed by qubit id.
-    """
-
-    def to_dict(self):
-        """
-        Create a JSON serializable representation of the noise model.
-
-        For example::
-
-            {
-                "isa_name": "example_qpu",
-                "gates": [
-                    # list of embedded dictionary representations of KrausModels here [...]
-                ]
-                "assignment_probs": {
-                    "0": [[.8, .1],
-                          [.2, .9]],
-                    "1": [[.9, .4],
-                          [.1, .6]],
-                }
-            }
-
-        :return: A dictionary representation of self.
-        :rtype: Dict[str,Any]
-        """
-        return {
-            "isa_name": self.isa_name,
-            "gates": [km.to_dict() for km in self.gates],
-            "assignment_probs": {str(qid): a.tolist() for qid, a in self.assignment_probs.items()},
-        }
-
-    @staticmethod
-    def from_dict(d):
-        """
-        Re-create the noise model from a dictionary representation.
-
-        :param Dict[str,Any] d: The dictionary representation.
-        :return: The restored noise model.
-        :rtype: NoiseModel
-        """
-        return NoiseModel(
-            isa_name=d["isa_name"],
-            gates=[KrausModel.from_dict(t) for t in d["gates"]],
-            assignment_probs={int(qid): np.array(a) for qid, a in d["assignment_probs"].items()},
-        )
-
-    def gates_by_name(self, name):
-        """
-        Return all defined noisy gates of a particular gate name.
-
-        :param str name: The gate name.
-        :return: A list of noise models representing that gate.
-        :rtype: Sequence[KrausModel]
-        """
-        return [g for g in self.gates if g.gate == name]
-
-    def __eq__(self, other):
-        return isinstance(other, NoiseModel) and self.to_dict() == other.to_dict()
-
-    def __neq__(self, other):
-        return not self.__eq__(other)
+    gates = []
+    for q in isa.qubits:
+        if q.dead:
+            continue
```

Comment:
we should mark this as a place we'll have to return to if we decide to handle dead qubit indexing differently, since this change (and i guess generally the integration of ISAs with pyQuil) means that it's not just the internal stuff that's going to need to get updated.

Reply:
Added a comment